streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -384,12 +384,13 @@ public GroupAssignment assign(final Cluster metadata, final GroupSubscription gr
                                 // if this topic is one of the sink topics of this topology,
                                 // use the maximum of all its source topic partitions as the number of partitions
                                 for (final String sourceTopicName : otherTopicsInfo.sourceTopics) {
-                                    final int numPartitionsCandidate;
+                                    int numPartitionsCandidate = 0;
                                     // It is possible the sourceTopic is another internal topic, i.e,
                                     // map().join().join(map())
-                                    if (repartitionTopicMetadata.containsKey(sourceTopicName)
```

Comment:
Here by checking if the source topic is a repartition topic **_and if_**  the number of partitions are present, we are expecting they might not be available.

However, if that is the case we will drop down to the `else` block and this results in throwing an `Exception`.  IMHO it seems that is not the intent of the logic, as we would always throw and `Exception` if **_any_** source topic (internal or otherwise) did not have a partition count available.